front/app/components/HookForm/SelectionChoiceSetup/index.tsx

```diff
@@ -0,0 +1,242 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { DndProvider } from 'react-dnd-cjs';
+import HTML5Backend from 'react-dnd-html5-backend-cjs';
+import { get } from 'lodash-es';
+
+// react hook form
+import { Controller, useFieldArray, useFormContext } from 'react-hook-form';
+
+// components
+import {
+  Box,
+  Label,
+  Button,
+  LocaleSwitcher,
+  Icon,
+  Input,
+} from '@citizenlab/cl2-component-library';
+import { SectionField } from 'components/admin/Section';
+import { List, SortableRow } from 'components/admin/ResourceList';
+import Error, { TFieldName } from 'components/UI/Error';
+
+// Typings
+import { Locale, CLError, RHFErrors } from 'typings';
+
+// utils
+import { isNilOrError } from 'utils/helperUtils';
+import { generateTempId } from 'components/FormBuilder/components/FormBuilderSettings/utils';
+
+interface Props {
+  name: string;
+  onSelectedLocaleChange?: (locale: Locale) => void;
+  locales: Locale[];
+  allowDeletingAllOptions?: boolean;
+  platformLocale: Locale;
+  fieldLabel: string | JSX.Element | null;
+  addButtonLabel: string | JSX.Element | undefined;
+}
+
+const SelectionChoiceSetup = ({
+  onSelectedLocaleChange,
+  name,
+  locales,
+  allowDeletingAllOptions = false,
+  platformLocale,
+  fieldLabel,
+  addButtonLabel,
+}: Props) => {
+  const {
+    control,
+    formState: { errors: formContextErrors },
+    setValue,
+    trigger,
+  } = useFormContext();
+  const [selectedLocale, setSelectedLocale] = useState<Locale | null>(
+    platformLocale
+  );
+
+  // Handles locale change
+  useEffect(() => {
```

Comment:
Do we have to change the locale if we change the platform locale? Doesn't seem super necessary IMO